### PR TITLE
integration, vendor: remove e2esetup dependency from code (#63)

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4709.

We were using e2esetup to create/delete tiller namespace in tests. We
don't want the dependency in the code though because e2esetup depends on
helmclient. This creates potential circular dependency problem and makes
vendoring hard.

### CODE REVIEW

File: integration/setup/k8s/setup.go [hunk 1]
```diff
@@ -13,6 +13,14 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
+/*
+	NOTE: This code is copied from e2esetup/k8s package to avoid circular
+	dependencies. All changes made there should be reflected in e2esetup.
+
+		See https://github.com/giantswarm/e2esetup/pull/13
+
+*/
```
Comment (kopiczko) [inline integration/setup/k8s/setup.go]: The diff is small because git thinks the file was moved from vendor directory. This is a full copy.

**Comment by xh3b4sd** [comment]
Why are we doing this? I understand this for the single constant in `e2e-harness` but this here goes a bit too far IMO. Why don't we have a common lib? Let's please not copy so much and try to "fix" it with a comment.

**Comment by kopiczko** [comment]
> Why are we doing this?

Circular dependencies. I explained that in the description.

> Why don't we have a common lib?

Maybe at some point that would be a good idea to have it. I'm not sure how should it look like ATM and what would be its scope. Things are changing in e2e and I don't want to constrain myself. This change debatable I agree. I'll try to do the upgrades with overrides.

**Comment by xh3b4sd** [comment]
If you like we can just hang and come up with some proper dependency graph. Or just create a common lib for that.

**Comment by xh3b4sd** [comment]
Cool. Just give me a shout.

**Comment by kopiczko** [comment]
I chatted with @xh3b4sd. We agreed it's better to just use `[[override]]` for `k8s.io` libs in `helmclient`. We are not sure what is the future of our testing libraries. We should figure that out first before extracting some random stuff (which may require e2e tests itself) to a separate repo.

**Pull request closed without merging**